src/libraries/Common/src/System/IO/FileSystem.DirectoryCreation.Windows.cs

```diff
@@ -34,39 +34,49 @@ public static unsafe void CreateDirectory(string fullPath, byte[]? securityDescr
             // isn't threadsafe.
 
             bool somepathexists = false;
-            int length = fullPath.Length;
+
+            ReadOnlySpan<char> fullPathSpan = fullPath;
+            int length = fullPathSpan.Length;
 
             // We need to trim the trailing slash or the code will try to create 2 directories of the same name.
-            if (length >= 2 && PathInternal.EndsInDirectorySeparator(fullPath.AsSpan()))
+            if (length >= 2 && PathInternal.IsDirectorySeparator(fullPathSpan[^1]))
             {
                 length--;
             }
 
-            int lengthRoot = PathInternal.GetRootLength(fullPath.AsSpan());
+            int lengthRoot = PathInternal.GetRootLength(fullPathSpan);
 
-            if (length > lengthRoot)
+            using (DisableMediaInsertionPrompt.Create())
```

Comment:
why add this -- was it missing?

Reply:
This is one of the main reasons why we use the Slim Method.
In the non-Slim method, we do this at the method level.

DisableMediaInsertionPrompt makes a kernel32 call at initialization and dispose, which we execute only once instead of calling `DirectoryExistsSlim`.

https://github.com/dotnet/runtime/blob/c926642522a3cc9723a7056f4f730ec2a39bf210/src/libraries/Common/src/System/IO/FileSystem.Attributes.Windows.cs#L54-L70